### PR TITLE
feat(components/forms): add field group component

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2212/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2212/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2212/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2212/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2212/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2212/playground/)

File: libs/components/forms/src/lib/modules/field-group/field-group.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  public labelHidden = false;
+
+  /**
+   * Whether the field group is stacked on another component. When specified, the appropriate
+   * vertical spacing is automatically added to the field group.
+   * @preview
+   */
+  @Input({ transform: booleanAttribute })
+  @HostBinding('class.sky-margin-stacked-xl')
+  public set stacked(value: boolean) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/field-group/field-group.component.ts]: Does this need to be a getter/setter?
```
@Input({ transform: booleanAttribute })
@HostBinding('class.sky-margin-stacked-xl')
public stacked = false;
```

File: libs/components/forms/testing/src/checkbox/checkbox-group-harness.ts [hunk 2]
```diff
@@ -45,15 +45,15 @@ export class SkyCheckboxGroupHarness extends SkyComponentHarness {
   }
 
   /**
-   * Gets the checkbox's label text. If the label is set via `labelText` and `labelHidden` is true,
+   * Gets the checkbox group's label text. If `labelHidden` is true
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/testing/src/checkbox/checkbox-group-harness.ts]: "...is true," (needs the comma)

File: libs/components/forms/src/lib/modules/field-group/field-group.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  /**
+   * Indicates whether to hide the `labelText`.
+   * @preview
+   */
+  @Input({ transform: booleanAttribute })
+  public labelHidden = false;
+
+  /**
+   * Whether the field group is stacked on another component. When specified, the appropriate
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/field-group/field-group.component.ts]: Story mentions this (and Adam has said it in the past), but it should be "...is stacked on another field group". Just so we're precise with our wording/design recommendations.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**